frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1531,23 +1541,17 @@ impl<T: Config> Pallet<T> {
 		//   inexpensive (1 read of an empty vector).
 		let _ = Self::finalize_signed_phase();
 		<QueuedSolution<T>>::take()
-			.map_or_else(
-				|| {
-					T::Fallback::elect()
-						.map_err(|fe| ElectionError::Fallback(fe))
-						.map(|supports| (supports, ElectionCompute::Fallback))
-				},
-				|ReadySolution { supports, compute, .. }| Ok((supports, compute)),
-			)
-			.map(|(supports, compute)| {
-				Self::deposit_event(Event::ElectionFinalized { election_compute: Some(compute) });
+			.ok_or(ElectionError::<T>::NothingQueued)
```

Comment:
This is a key line of code that I hadn't thought of before. And that's why i removed all the map functions .

Reply:
To be realistic, this error is actually pointless, since it is never returned (we always return the fallback error if it fails), but yeah it is a needed step to go from Option to Error